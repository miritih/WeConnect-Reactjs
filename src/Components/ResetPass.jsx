import React from 'react';
import NavBar from './NavBar';
import ProfileNav from './ProfileNav';
import ResetPassForm from './forms/ResetPassForm';
class ResetPass extends React.Component {
  render() {
    return (
      <div>
        <NavBar 
          history={this.props.history} 
          location={this.props.location} 
        />
      <div className="container profile">
        <div className="card">
          <div className="card-body">
            <div className="row">
            <div className="col-sm-3">
             <ProfileNav />
            </div>
            <div className="col-sm-9"><h5>Reset Password</h5>
              <hr/>
              <ResetPassForm />
              </div>
            </div>
          </div>
          </div>
        </div>
      </div>
    );
  }
}
export default ResetPass;
