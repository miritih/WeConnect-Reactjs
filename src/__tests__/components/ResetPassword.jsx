import React from 'react';
import ResetPass from '../../Components/ResetPass';
import { MemoryRouter as Router } from 'react-router-dom';
// import renderer for snapshot testing
import renderer from 'react-test-renderer';

describe('ResetPassword', () => {
  const pathname = {
    pathname: "/"
  };

  it('renders without crashing', () => {
    const rendered = renderer.create(
      <Router><ResetPass location={pathname} /></Router>
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
