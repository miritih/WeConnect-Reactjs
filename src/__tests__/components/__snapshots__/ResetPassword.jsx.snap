// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPassword renders without crashing 1`] = `
<div>
  <div>
    <nav
      className="navbar navbar-expand-lg navhome navbar-light bg-transparent fixed-top"
    >
      <div
        className="container"
      >
        <a
          className="navbar-brand"
          href="/"
          onClick={[Function]}
        >
          <span
            className="white"
          >
            WeConnect
          </span>
        </a>
        <button
          aria-controls="topnav"
          aria-expanded="false"
          aria-label="Toggle navigation"
          className="navbar-toggler"
          data-target="#topnav"
          data-toggle="collapse"
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <div
          className="collapse navbar-collapse"
          id="topnav"
        >
          <ul
            className="navbar-nav ml-auto white"
          >
            <li
              className="nav-item"
            >
              <a
                className="nav-link"
                href="/"
                onClick={[Function]}
              >
                Home
              </a>
            </li>
            <li
              className="nav-item"
            >
              <a
                className="nav-link"
                href="/register"
                onClick={[Function]}
              >
                Signup
              </a>
            </li>
            <li
              className="nav-item"
            >
              <a
                className="nav-link"
                href="/login"
                onClick={[Function]}
              >
                Login
              </a>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  </div>
  <div
    className="container profile"
  >
    <div
      className="card"
    >
      <div
        className="card-body"
      >
        <div
          className="row"
        >
          <div
            className="col-sm-3"
          >
            <ul
              className="list-group"
            >
              <li
                className="list-group-item list-group-item-dark"
              >
                Personal settings
              </li>
              <li
                className="list-group-item"
              >
                <a
                  className="dropdown-item"
                  href="/profile/update"
                  onClick={[Function]}
                >
                  Edit Profile
                </a>
              </li>
              <li
                className="list-group-item"
              >
                <a
                  className="dropdown-item"
                  href="/profile/admin"
                  onClick={[Function]}
                >
                  Change Password
                </a>
              </li>
              <li
                className="list-group-item"
              >
                <a
                  className="dropdown-item"
                  href="/profile/business"
                  onClick={[Function]}
                >
                  Your Businesses
                </a>
              </li>
            </ul>
          </div>
          <div
            className="col-sm-9"
          >
            <h5>
              Reset Password
            </h5>
            <hr />
            <form
              onSubmit={[Function]}
            >
              <label
                htmlFor="old_password"
              >
                Old Password
              </label>
              <input
                className="form-control "
                name="old_password"
                onChange={[Function]}
                required={true}
                type="password"
                value=""
              />
              <br />
              <label
                htmlFor="new_password"
              >
                New Password
              </label>
              <input
                className="form-control "
                name="new_password"
                onChange={[Function]}
                required={true}
                type="password"
                value=""
              />
              <br />
              <label
                htmlFor="confirm_password"
              >
                Confirm Password
              </label>
              <input
                className="form-control "
                name="confirm_password"
                onChange={[Function]}
                type="password"
                value=""
              />
              <br />
              <br />
              <button
                className="btn btn-primary"
                type="submit"
              >
                Reset Password
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
